### DISCUSSION TITLE
Micropython port

### DISCUSSION THREAD

**Discussion by matiasmorant** [marcel-licence/esp32_basic_synth#93] [General]
C is such a pain.
Would micropython be suitable for this project?
I can work on porting next year.

**Comment by marcel-licence** [comment]
Sounds like an interesting idea. I am not sure if a port would be possible. Do you know if there is any "runtime" available for the ESP32?
I could imagine that having an interpreter in between could cause performance issues. 
Maybe the sound-core could be left as it is and only the controlling part could be done in python.
What do you think about it?

**Reply by matiasmorant** [reply]
I know this:

- micropython & circuitpython are supported on esp32
- micropython is much slower than C, faster than real python.
- the program running on the chip will be the python interpreter. It will read a user python file on startup and run it.

I don't know:

- can micropython be compiled so you don't need the interpreter? This would make it faster
- calling C from micropython seemed complicated.
